File: python/aoc/day_one.py
import os
from pathlib import Path
from typing import List


def day_one(file_path: Path):
    lines = read_file(file_path)
    max_calorie_elf, max_calories = find_max_calorie_elf(lines)
    print("Max calorie elf: {}, max_calories: {}".format(max_calorie_elf, max_calories))


def read_file(file_path: Path):
    with open(file_path, 'r') as f:
        lines = f.readlines()
        return lines


def find_max_calorie_elf(lines: List[str]):
    max_calorie_elf = 1
    max_calories = 0
    current_calories = 0
    current_elf = 1
    for line in lines:
        line = line.replace("\n", "")
        if len(line) == 0:
            if current_calories > max_calories:
                max_calories = current_calories
                max_calorie_elf = current_elf

            current_elf += 1
            current_calories = 0
        else:
            current_calories += int(line)

    return max_calorie_elf, max_calories

File: go/aoc/day_one.go
package main

import (
	"fmt"
	"strconv"
)

func day_one(file_path string) {
	lines, err := read_file_to_array(file_path)
	if err != nil {
		fmt.Println(err)
	}
	max_calories := 0
	max_calorie_elf := 1
	current_calories := 0
	current_elf := 1

	for _, line := range lines {
		if line == "" {
			if current_calories > max_calories {
				max_calories = current_calories
				max_calorie_elf = current_elf
			}
			current_calories = 0
			current_elf += 1
		} else {
			calorie, _ := strconv.Atoi(line)
			current_calories += calorie
		}
	}
	fmt.Println("Elf", max_calorie_elf, "has the most calories:", max_calories)

}
